operator/pkg/gateway-api/cell.go

```diff
@@ -119,17 +125,14 @@ func initGatewayAPIController(params gatewayAPIParams) error {
 		return err
 	}
 
-	params.Logger.Info("Checking for required GatewayAPI resources", "requiredGVK", requiredGVK)
-	if err := checkRequiredCRDs(context.Background(), params.K8sClient); err != nil {
+	params.Logger.Info("Checking for required and optional GatewayAPI resources", "requiredGVK", requiredGVKs, "optionalGVK", optionalGVKs)
+	installedKinds, err := checkCRDs(context.Background(), params.K8sClient, params.Logger)
```

Comment:
IMO it would be more expressive (and the function better testable) if we would pass `requiredGVKs` & `optionalGVKs` (that are defined as consts) as parameters to the function.

Reply:
Because we need `requiredGVKs` and `optionalGVKs` to be `[]schema.GroupVersionKind`, they can't be `const`. But I agree that passing them in to this function makes it way more testable.